### DISCUSSION TITLE
Manifold as Header Library

### DISCUSSION THREAD

**Discussion by pentacular** [elalish/manifold#167] [Ideas]
It would be nice to be able to use manifold as a header library.

Are there any downside or opposition to restructuring it like that?

**Comment by pca006132** [comment]
Not sure if this is useful. I think header only library is useful when most of the functions are template functions, but manifold don't have much template functions. Also, this will not simplify the build process as users will have to set the compiler options for their code that includes the manifold headers.

**Comment by pentacular** [comment]
It also helps with pruning code that you don't use.

**Comment by pca006132** [comment]
If you want to prune functions that you don't use, maybe you can look at some linker options, e.g. `--gc-sections ` with `-ffunction-sections`.

**Comment by elalish** [comment]
Also, do you have a sense of how much space you could save by pruning? Most of the code base is used just to create the Boolean operations; I think even if you used nothing else, you couldn't really prune that much out.

**Comment by fire** [comment]
Header libraries are easier to use, but if it makes it harder to code it's not worth it. I have no real opinion.

**Comment by elalish** [comment]
I'm converting this to a discussion since I don't think there's anything actionable here for now.
